augur/filter/_run.py

```diff
@@ -364,39 +371,35 @@ def run(args):
     # Force inclusion of specific strains after filtering and subsampling.
     valid_strains = valid_strains | all_sequences_to_include
 
-    # Write output starting with sequences, if they've been requested. It is
-    # possible for the input sequences and sequence index to be out of sync
-    # (e.g., the index is a superset of the given sequences input), so we need
-    # to update the set of strains to keep based on which strains are actually
-    # available.
-    if is_vcf:
-        if args.output_sequences:
-            # Get the samples to be deleted, not to keep, for VCF
-            dropped_samps = list(sequence_strains - valid_strains)
-            write_vcf(args.sequences, args.output_sequences, dropped_samps)
-    elif args.sequences:
-        # If the user requested sequence output, stream to disk all sequences
-        # that passed all filters to avoid reading sequences into memory first.
-        # Even if we aren't emitting sequences, we check for duplicates and
-        # track the observed strain names in the sequence file as part of the
-        # single pass to allow comparison with the provided sequence index.
-        observed_sequence_strains = set()
-        duplicates = set()
-        with open_file(args.output_sequences, "wt") if args.output_sequences else nullcontext() as output_handle:
-            for sequence in read_sequences(args.sequences):
-                if sequence.id in observed_sequence_strains:
-                    duplicates.add(sequence.id)
-
-                observed_sequence_strains.add(sequence.id)
-
-                if args.output_sequences:
-                    if sequence.id in valid_strains:
-                        write_sequences(sequence, output_handle, 'fasta')
-
-        if duplicates:
+    # Write requested outputs.
+
+    # Write a strains file if explicitly requested or if sequence output is requested.
+    strains_file = None
+    if args.output_strains or args.output_sequences:
+        with NamedTemporaryFile(mode="w+", delete=False) as temp_file:
+            strains_file = temp_file.name
+            print_debug(f"Writing strains to temporary file {temp_file.name!r}…")
+            for strain in valid_strains:
+                temp_file.write(f"{strain}\n")
+
+        if args.output_strains:
+            print_debug(f"Copying strains to {args.output_strains!r}…")
+            copyfile(strains_file, args.output_strains)
+
+    # For non-VCF sequence inputs, check ids for duplicates and compare against sequence index.
+    if args.sequences and not is_vcf:
+        print_debug(f"Reading sequences from {args.sequences!r}…")
+
+        try:
+            observed_sequence_strains = read_sequence_ids(args.sequences, error_on_duplicates=True)
```

Comment:
Ah, just realized that all the documentation I've read around use of `seqkit` was for `augur merge`. I don't think there's documentation in `augur filter` of the requirement to have `seqkit` installed. 

Along with docs, I think we could also check for installation of `seqkit` in `validate_arguments` (similar to how we [check for vcftools](https://github.com/nextstrain/augur/blob/49197e07f43b98f1dad18fb36cc82dbf648957c0/augur/filter/validate_arguments.py#L37-L42)). Otherwise, `augur filter` might do all the work for filtering just to error out here because `seqkit` isn't available.